loader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/loader/LoaderCli.scala

```diff
@@ -10,26 +10,28 @@
  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
  */
-package com.snowplowanalytics.snowplow.storage.bigquery
-package loader
+package com.snowplowanalytics.snowplow.storage.bigquery.loader
 
-import cats.effect.{Clock, IO}
+import cats.effect.IO
 import com.spotify.scio.Args
 
-import com.snowplowanalytics.snowplow.storage.bigquery.common.Config._
+import com.snowplowanalytics.snowplow.storage.bigquery.common.config.CliConfig.{
+  decodeBase64Hocon,
+  decodeBase64Json,
+  validateResolverJson
+}
+import com.snowplowanalytics.snowplow.storage.bigquery.common.config.CliConfig.Environment.LoaderEnvironment
 
 /**
-  * Loader specific CLI configuration
-  * Unlike Mutator, required --key=value format and ignores unknown options (for Dataflow)
+  * Loader specific CLI configuration.
+  * Requires --key=value format and ignores unknown options (for Dataflow)
   */
 object LoaderCli {
-  implicit private val privateIoClock: Clock[IO] =
-    Clock.create[IO]
-
-  def parse(args: Args): Either[Throwable, Environment] =
+  def parse(args: Args): Either[Throwable, LoaderEnvironment] =
     for {
-      c <- decodeBase64Json(args("config"))
+      c <- decodeBase64Hocon(args("config"))
       r <- decodeBase64Json(args("resolver"))
-      e <- transform[IO](EnvironmentConfig(r, c)).value.unsafeRunSync()
+      _ = validateResolverJson[IO](r).value.unsafeRunSync()
```

Comment:
Have you checked whether this correctly rejects an invalid resolver json?  Because I think this line should be `<-` instead of `=` i.e.

```
_ <- validateResolverJson[IO](r).value.unsafeRunSync()
```

(you will need to leftMap the either you to a Throwable instead of a String)

Can you get it to work with the `[Id]` effect type instead of `[IO]`?  Then you don't need the ugly unsafeRunSync.

Reply:
These are good points and I'll pay attention to them but I'm also thinking now that this check should probably be moved out of the modules and into the `decodeBase64Json` function in the main `Cli` file.